internal/metrics/metrics.go

```diff
@@ -108,6 +107,65 @@ func (m *recordReadyMetric) Publish() {
 	RecordReady.WithLabelValues(m.name, m.namespace, m.rootHost, strconv.FormatBool(m.delegating)).Set(gauge)
 }
 
+type RemoteClusterCollector struct {
+	Provider *kubeconfigprovider.Provider
+}
+
+func (c *RemoteClusterCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- multiClusterCount
+}
+
+func (c *RemoteClusterCollector) Collect(ch chan<- prometheus.Metric) {
+	ch <- prometheus.MustNewConstMetric(
+		multiClusterCount,
+		prometheus.GaugeValue,
+		float64(len(c.Provider.ListClusters())),
+	)
+}
+
+type LocalCollector struct {
+	Ctx context.Context
+	Mgr manager.Manager
+}
+
+func (c *LocalCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- authoritativeRecordSpecInfo
+}
+
+func (c *LocalCollector) Collect(ch chan<- prometheus.Metric) {
+
+	cl := c.Mgr.GetClient()
+	logger := c.Mgr.GetLogger()
+
+	listOptions := &client.ListOptions{}
+	dnsRecordList := &v1alpha1.DNSRecordList{}
+
+	err := cl.List(c.Ctx, dnsRecordList, listOptions)
+	if err != nil {
+		logger.Info("failed to list dns records", "status", "error")
+	}
+
+	for _, record := range dnsRecordList.Items {
```

Comment:
So this is where other metrics will be registered once we "move" the rest of the metrics to this pattern?